src/data/cam_thermo.F90

```diff
@@ -1649,78 +1682,96 @@ subroutine get_hydrostatic_energy_1hd(tracer, pdel, cp_or_cv, U, V, T,     &
+                    0.5_kind_phys * (U(idx, kdx)**2 + V(idx, kdx)**2)) * rga
                se_vint(idx) = se_vint(idx) + (T(idx, kdx) *                   &
-                    cp_or_cv(idx, kdx) * pdel(idx, kdx) / gravit)
-               wv_vint(idx) = wv_vint(idx) + (tracer(idx, kdx, wvidx) *       &
-                    pdel(idx, kdx) / gravit)
+                    cp_or_cv(idx, kdx) * pdel(idx, kdx) * rga)
+               po_vint(idx) =  po_vint(idx)+pdel(idx, kdx)
+
             end do
          end do
          do idx = 1, SIZE(tracer, 1)
-            se_vint(idx) = se_vint(idx) + (phis(idx) * ps(idx) / gravit)
+            po_vint(idx) =  (phis(idx) * po_vint(idx) * rga)
          end do
-      case(vc_height)
-         if (.not. present(z_mid)) then
-            write(iulog, *) subname,                                          &
-                 ' z_mid must be present for height vertical coordinate'
-            call endrun(subname//': z_mid must be present for height '//      &
-                 'vertical coordinate')
+      case(ENERGY_FORMULA_DYCORE_MPAS)
+         if (.not. present(phis)) then
+            write(iulog, *) subname, ' phis must be present for ',     &
+                 'MPAS energy formula'
+            call endrun(subname//':  phis must be present for '//      &
+                 'MPAS energy formula')
          end if
-         ke_vint = 0._kind_phys
-         se_vint = 0._kind_phys
-         wv_vint = 0._kind_phys
+         po_vint = 0._kind_phys
          do kdx = 1, SIZE(tracer, 2)
             do idx = 1, SIZE(tracer, 1)
                ke_vint(idx) = ke_vint(idx) + (pdel(idx, kdx) *                &
-                    0.5_kind_phys * (U(idx, kdx)**2 + V(idx, kdx)**2) / gravit)
+                    0.5_kind_phys * (U(idx, kdx)**2 + V(idx, kdx)**2) * rga)
                se_vint(idx) = se_vint(idx) + (T(idx, kdx) *                   &
-                    cp_or_cv(idx, kdx) * pdel(idx, kdx) / gravit)
+                    cp_or_cv(idx, kdx) * pdel(idx, kdx) * rga)
                ! z_mid is height above ground
-               se_vint(idx) = se_vint(idx) + (z_mid(idx, kdx) +               &
-                    phis(idx) / gravit) * pdel(idx, kdx)
-               wv_vint(idx) = wv_vint(idx) + (tracer(idx, kdx, wvidx) *       &
-                    pdel(idx, kdx) / gravit)
+               po_vint(idx) = po_vint(idx) + (z_mid(idx, kdx) +               &
+                    phis(idx) * rga) * pdel(idx, kdx)
             end do
          end do
       case default
-         write(iulog, *) subname, ' vertical coordinate not supported: ', vcoord
-         call endrun(subname//': vertical coordinate not supported')
+         write(iulog, *) subname, ' energy formula not supported: ', vcoord
+         call endrun(subname//': energy formula not supported')
       end select
       if (present(te)) then
-         te  = se_vint + ke_vint
+         te  = se_vint + po_vint + ke_vint
```

Comment:
Is this a bug fix? (old code only has ```te = se_vint + ke_vint```)

Reply:
This is to match current CAM's implementation of `get_hydrostatic_energy`. It was introduced in: https://github.com/ESCOMP/CAM/commit/d11b9f1d5d56f2d24ad2595288827a52a2350f91#diff-a8346d03f9c54a04be59b7d7bc88931a7669861f65f812ec27c23c71beb66705

described as
> separate "static energy" into enthalpy and PHIS energy

and included in cam6_3_109: https://github.com/ESCOMP/CAM/pull/761